### PR TITLE
refactor(interface): clean up

### PR DESCRIPTION
**Pull request by marcoscaceres**
- Add a few sections headings
- fold data dict into `ShareEvent`
- remove void reject(DOMException error);

### CODE REVIEW

File: docs/interface.md [hunk 1]
```diff
@@ -50,9 +55,6 @@ agents. It is easier to add such a method later than remove it.
 
 ## Event handlers
 
-Handlers **must** have a registered [service
```
Comment (marcoscaceres) [inline docs/interface.md]: This is an authoring requirement, so doesn't belong here. For this to work, obviously, the developer would have had to subscribe the SW as a target (though I don't think they necessarily need to have the event handler set up, as that can happen dynamically when the SW wakes up).  

For example: 

```JS
if(theDayIsMonday){
   self.addEventListener("share", ()=>{
   });
}
```

Comment (marcoscaceres) [inline docs/interface.md]: To be clear, what tells the UA to wake up the SW on share the `ShareManager` subscription, not the registration of the event handler. Registration of an event handler must be side-effects free.

Comment (mgiuca) [inline docs/interface.md]: I'm not clear on why that necessitates removing this sentence.

File: docs/interface.md [hunk 2]
```diff
@@ -63,20 +65,14 @@ partial interface WorkerGlobalScope {
 };
 
 interface ShareEvent : ExtendableEvent {
-  readonly attribute ShareData data;
-
-  void reject(DOMException error);
-};
-
-dictionary ShareData {
```
Comment (mgiuca) [inline docs/interface.md]: Not sure about this change. It makes sense to not have a nesting level if `data` is the only field in `ShareEvent`. But I also wanted to a) share `ShareData` between this and the [Web Share spec](https://github.com/WICG/web-share/blob/master/docs/interface.md), and b) cleanly separate the data struct from the other methods and fields of `Event`.

File: docs/interface.md [hunk 3]
```diff
@@ -63,20 +65,14 @@ partial interface WorkerGlobalScope {
 };
 
 interface ShareEvent : ExtendableEvent {
-  readonly attribute ShareData data;
-
-  void reject(DOMException error);
```
Comment (mgiuca) [inline docs/interface.md]: Why remove the `reject` method?

File: docs/interface.md [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
+**Note**: The Web Share Target API is a proposal of the [Ballista
+project](https://github.com/chromium/ballista), which aims to explore
+website-to-website and website-to-native interoperability.
+
+## Dependencies
+
 This API requires the user agent to support both [service
 workers](https://www.w3.org/TR/service-workers/) and [web app
 manifests](https://www.w3.org/TR/appmanifest/). The [Web Share
 API](https://github.com/mgiuca/web-share) is not required, but recommended.
```
Comment (mgiuca) [inline docs/interface.md]: s/mgiuca/WICG.

**Comment by marcoscaceres** [comment]
This is way outdated. Closing.

**Pull request closed without merging**